pkg/scheduler/framework/v1alpha1/framework.go

```diff
@@ -222,6 +237,27 @@ func (f *framework) RunPrefilterPlugins(
 	return nil
 }
 
+// RunFilterPlugins runs the set of configured filter plugins for pod on each candidate host.
+// If any of these plugins returns error, the node is not suitable for running pod.
+// Meanwhile, the failure message and status are set for each non-candidate host.
+func (f *framework) RunFilterPlugins(pc *PluginContext,
+	pod *v1.Pod, nodeName string) *Status {
+
+	nodeInfo, exist := f.nodeInfoSnapshot.NodeInfoMap[nodeName]
+	if !exist {
+		return NewStatus(Error, "Fail to find NodeInfo in snapshot")
+	}
+
+	for _, p := range f.filterPlugins {
+		status := p.Filter(pc, pod, nodeInfo)
+		if !status.IsSuccess() {
```

Comment:
we should distinguish between success/unschedulable and all other statuses. if success or unschedulable, then we return directly, otherwise we should probably convert the status to Error like we do for other extension points.

Reply:
Filter plugins are special, as their returned message is shown to the user. For example, if there is not enough memory on a node, we put a message in pod status which is similar to "10 nodes insufficient memory". The part that formats the message for end user is in the scheduler code in `predicates.NewFailureReason()`, but the "reason" part of the message (which is "insufficient memory" in the above example) is important. We should not alter the `status.Message` here for that reason. So, returning `status` as is, is fine IMO. We may want to check for error and log errors though.

Reply:
Message is logged once the plugin returns failure.

Reply:
@bsalamat we can keep the message that plugin returned when converting the code to Error: 

return NewStatus(Error, status.Message())

I just want to make sure we don't bubble statuses that don't make sense for a filter plugin, like "Wait".

Reply:
+1 to @ahg-g's suggestion. It makes sense to convert all codes other than "Unschedulable" to an `Error`

Reply:
@YoubingLi this hasn't been addressed yet. I left a comment with the suggestion.